### PR TITLE
ci: add manual workflow for release

### LINKED ISSUE #1242: 🤖 - Create a release branch by bot
### Description

Analogue: https://github.com/conventional-changelog/standard-version/pull/867

image

+ Add release on github

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

Closes #1242 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          token: ${{ secrets.TINKOFF_BOT_PAT }}
+
+      - name: Setup Node.js and Cache
+        uses: ./.github/actions/nodejs
+
+      - name: Run release
+        run: |
+          git config user.name "tinkoff-bot"
+          git config --global push.followTags true
+          npm run release -- --release-as ${{ github.event.inputs.mode }}`
```
Comment (splincode) [inline .github/workflows/create-release.yml:34]:
```
}}`
```

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          npm run prettier
+          git add .
+          git commit --all --amend --no-edit --no-verify
+          NEW_VERSION=$(node -p "require('./package.json').version")
+          echo "new_version = $NEW_VERSION" >> $GITHUB_ENV
```
Comment (splincode) [inline .github/workflows/create-release.yml:39]: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions
please don't use file overwriting with variables, it's bad practice

image

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        default: 'minor'
+        options:
+          - patch
+          - minor
+          - major
+
+jobs:
+  run-release:
+    name: Demo
+    runs-on: ubuntu-latest
```
Comment (splincode) [inline .github/workflows/create-release.yml:19]: `if: ${{ github.ref == 'refs/heads/main' }}`

File: .github/workflows/create-release.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        required: true
+        default: 'minor'
+        options:
+          - patch
+          - minor
+          - major
+
+jobs:
+  run-release:
+    name: Demo
```
Comment (splincode) [inline .github/workflows/create-release.yml:18]: `name: Release`

File: .github/workflows/create-release.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          git commit --all --amend --no-edit --no-verify
+          NEW_VERSION=$(node -p "require('./package.json').version")
+          echo "new_version = $NEW_VERSION" >> $GITHUB_ENV
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.TINKOFF_BOT_PAT }}
+          branch: release/{{env.new_version}}
+          title: 'Release/${{env.new_version}}'
```
Comment (splincode) [inline .github/workflows/create-release.yml:46]:
```yml
  labels: pending
```

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (vladimirpotekhin) [reply .github/workflows/create-release.yml]: thanks

**Review by splincode — APPROVED** [review]

**Comment by splincode** [comment]
Please, rename PR to `ci: add manual workflow for release`

File: scripts/release.js [hunk 6]
```diff
@@ -14,10 +14,10 @@ execSync(`git checkout -b release/${newVersion}`, options);
 execSync(`npm run release -- --release-as ${mode}`, options);
 
 execSync(`npm run prettier`, options);
-execSync(`git add . && git commit --all --amend --no-edit --no-verify`, options);
+execSync(`git commit -m 'chore: run prettier' --no-verify`);
 
 checkChangelog().then(() => {
-    execSync(`git add . && git commit --all --amend --no-edit --no-verify`, options);
+    execSync(`git commit -m 'chore(changelog): fix incorrect generated logs'`);
```
Comment (splincode) [inline scripts/release.js:20]: `--no-verify`

**Review by IKatsuba — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**